nix/rust.nix

```diff
@@ -87,6 +87,35 @@ in {
     doCheck = false;
   };
 
+  kimchi_stubs_static_lib = let
+    toolchain = rustChannelFromToolchainFileOf
+      # Using the same toolchain which is used by the local stubs crate
+      ../src/lib/crypto/kimchi_bindings/stubs/rust-toolchain.toml;
+    rust_platform = rustPlatformFor toolchain.rust;
+  in rust_platform.buildRustPackage {
+    pname = "kimchi_stubs_static_lib";
+    version = "0.1.0";
+    src = final.lib.sourceByRegex ../src [
+      "^lib(/crypto(/proof-systems(/.*)?)?)?$"
+    ];
+    sourceRoot = "source/lib/crypto/proof-systems";
+    nativeBuildInputs = [ final.ocamlPackages_mina.ocaml ];
+    buildInputs = with final; lib.optional stdenv.isDarwin libiconv;
+    cargoLock = let fixupLockFile = path: builtins.readFile path;
+    in {
+      lockFileContents =
+        fixupLockFile ../src/lib/crypto/proof-systems/Cargo.lock;
+    };
+    buildPhase = ''
+      cargo build -p kimchi-stubs --release --lib --all-features
```

Comment:
I mostly checked this, and it LGTM.

Reply:
Note that `--all-features` does not seem to be required, see
```
    (run
     cargo
     build
     -p
     kimchi-stubs
     --release
     --target-dir
     %{read:../dune-build-root}/cargo_kimchi_stubs
     --offline))
```

I'm wondering also if we could merge with the other variable, `MARLIN_PLONK_STUBS`.
I am fine with a follow-up for this, and I would also include a renaming of `MARLIN_PLONK_STUBS` into something more correct as now, like `KIMCHI_STUBS`. Marlin refers to an old proof systems Mina was using in the past.